### PR TITLE
New wizard spell - Corpse Explosion

### PR DESCRIPTION
**Pull request by McRamon**
## What Does This PR Do
Adds a new wizard spell, corpse explosion. The spell gibs a dead mob, heavily damaging mobs around them and applying similar to changelign last resort effects on mob visuals. Caster is immune to it.
Currently deals:
Knock down, 6 seconds confused and eye blurry on human mobs, 6 seconds weaken on silicons and applies 85 brute damage to all mobs.
Scales with range, at 2 tile range deals roughly 40 damage
Cooldown is similar to fireball, as in pretty fast since it also requires a condition of a corpse to be casted.
Numbers can change depending on balance team input

## Why It's Good For The Game
Its Diablo 2 reference

## Images of changes
image
dreamseeker_PaX2bI90Rn
my corpse oh no

## Testing

Detonated skrells and Coffees 

## Changelog
:cl: McRamon, DGamerL
add: New wizard spell - corpse explosion
/:cl:

### CODE REVIEW

**Comment by Inferno1882** [comment]
I'm no Balance Team member, but a 100 damage explosion sounds like immediate Medbay Destroyer with little actual effort.

File: code/datums/spells/wizard_spells.dm [hunk 1]
```diff
@@ -478,3 +478,54 @@
 	if(isliving(user))
 		var/mob/living/U = user
 		U.IgniteMob()
+
+/obj/effect/proc_holder/spell/corpse_explosion
+	name = "Corpse Explosion"
+	desc = "Fills a corpse with energy causing it to explode violently"
+	school = "evocation"
+	base_cooldown = 50
```
Comment (DGamerL) [inline code/datums/spells/wizard_spells.dm:486]:
```suggestion
	base_cooldown = 5 SECONDS
```

File: code/datums/spells/wizard_spells.dm [hunk 2]
```diff
@@ -478,3 +478,54 @@
...
+	T.click_radius = 0
+	T.try_auto_target = FALSE
+	T.allowed_type = /mob/living
+	return T
+
+/obj/effect/proc_holder/spell/corpse_explosion/cast(list/targets, mob/user)
+	var/mob/living/target = targets[1]
+	var/turf/corpse_turf = get_turf(target)
+	if(!target || target.stat == !DEAD)
+		return
```
Comment (DGamerL) [inline code/datums/spells/wizard_spells.dm:505]:
```suggestion
	var/mob/living/target = targets[1]
	if(!target || target.stat == !DEAD)
		return
	var/turf/corpse_turf = get_turf(target)
```

File: code/datums/spells/wizard_spells.dm [hunk 3]
```diff
@@ -478,3 +478,54 @@
...
+	new /obj/effect/temp_visual/corpse_explosion(get_turf(target))
+	target.gib()
+	explosion(get_turf(target), 0, 0, 0, 0, silent = TRUE,breach = FALSE)
```
Comment (DGamerL) [inline code/datums/spells/wizard_spells.dm:508]:
```suggestion
	explosion(get_turf(target), 0, 0, 0, 0, silent = TRUE, breach = FALSE)
```

Comment (DGamerL) [inline code/datums/spells/wizard_spells.dm:508]:
```suggestion
	explosion(corpse_turf, 0, 0, 0, 0, silent = TRUE,breach = FALSE)
```

File: code/datums/spells/wizard_spells.dm [hunk 4]
```diff
@@ -478,3 +478,54 @@
...
+	for(var/mob/living/carbon/human/H in range(4, corpse_turf))
+		if(H == usr)
```
Comment (DGamerL) [inline code/datums/spells/wizard_spells.dm:510]:
```suggestion
		if(H == user)
```

File: code/datums/spells/wizard_spells.dm [hunk 5]
```diff
@@ -478,3 +478,54 @@
...
+		H.AdjustConfused(6 SECONDS / range)
+	for(var/mob/living/silicon/S in range(4, corpse_turf))
+		to_chat(S, "<span class='userdanger'>Your sensors are disabled and carapace ripped apart by the violent dark magic!</span>")
+		var/range = get_dist_euclidian(S, corpse_turf)
+		S.Weaken(6 SECONDS / range)
+		S.apply_damage(85 / range, BRUTE)
+	for(var/mob/living/M in range(1, corpse_turf))
+		if(M == /mob/living/silicon)
+			return
+		if(M == /mob/living/carbon/human)
```
Comment (DGamerL) [inline code/datums/spells/wizard_spells.dm:527]:
```suggestion
		if(ishuman(M))
```

File: code/datums/spells/wizard_spells.dm [hunk 6]
```diff
@@ -478,3 +478,54 @@
...
+		H.EyeBlurry(40 SECONDS / range)
+		H.apply_damage(85 / range, BRUTE)
+		H.AdjustConfused(6 SECONDS / range)
+	for(var/mob/living/silicon/S in range(4, corpse_turf))
+		to_chat(S, "<span class='userdanger'>Your sensors are disabled and carapace ripped apart by the violent dark magic!</span>")
+		var/range = get_dist_euclidian(S, corpse_turf)
+		S.Weaken(6 SECONDS / range)
+		S.apply_damage(85 / range, BRUTE)
+	for(var/mob/living/M in range(1, corpse_turf))
+		if(M == /mob/living/silicon)
```
Comment (DGamerL) [inline code/datums/spells/wizard_spells.dm:525]:
```suggestion
		if(issilicon(M))
```

File: code/datums/spells/wizard_spells.dm [hunk 7]
```diff
@@ -478,3 +478,54 @@
...
+	var/mob/living/target = targets[1]
+	if(!target || target.stat == !DEAD)
+		return
+	var/turf/corpse_turf = get_turf(target)
+	new /obj/effect/temp_visual/corpse_explosion(get_turf(target))
+	target.gib()
+	explosion(corpse_turf, 0, 0, 0, 0, silent = TRUE, breach = FALSE)
+	for(var/mob/living/carbon/human/H in range(4, corpse_turf))
+		if(H == user)
+			return
```
Comment (DGamerL) [inline code/datums/spells/wizard_spells.dm:511]:
```suggestion
			continue 
```

File: code/datums/spells/wizard_spells.dm [hunk 8]
```diff
@@ -478,3 +478,54 @@
...
+		H.apply_damage(85 / range, BRUTE)
+		H.AdjustConfused(6 SECONDS / range)
+	for(var/mob/living/silicon/S in range(4, corpse_turf))
+		to_chat(S, "<span class='userdanger'>Your sensors are disabled and carapace ripped apart by the violent dark magic!</span>")
+		var/range = get_dist_euclidian(S, corpse_turf)
+		S.Weaken(6 SECONDS / range)
+		S.apply_damage(85 / range, BRUTE)
+	for(var/mob/living/M in range(1, corpse_turf))
+		if(issilicon(M))
+			return
```
Comment (DGamerL) [inline code/datums/spells/wizard_spells.dm:526]:
```suggestion
			continue 
```

File: code/datums/spells/wizard_spells.dm [hunk 9]
```diff
@@ -478,3 +478,54 @@
...
+		if(ishuman(M))
+			return
```
Comment (DGamerL) [inline code/datums/spells/wizard_spells.dm:528]:
```suggestion
			continue 
```

File: code/datums/spells/wizard_spells.dm [hunk 10]
```diff
@@ -478,3 +478,48 @@
 	if(isliving(user))
 		var/mob/living/U = user
 		U.IgniteMob()
+
+/obj/effect/proc_holder/spell/corpse_explosion
+	name = "Corpse Explosion"
+	desc = "Fills a corpse with energy causing it to explode violently"
```
Comment (Henri215) [inline code/datums/spells/wizard_spells.dm:484]:
```suggestion
	desc = "Fills a corpse with energy, causing it to explode violently."
```

File: code/datums/spells/wizard_spells.dm [hunk 11]
```diff
@@ -478,3 +478,48 @@
...
+	school = "evocation"
+	base_cooldown = 5 SECONDS
+	clothes_req = TRUE
+	invocation = "JAH ITH BER"
+	invocation_type = "whisper"
+	selection_activated_message = "<span class='notice'>You prepare to detonate a corpse. Click on a target to cast the spell.</span>"
+	selection_deactivated_message = "<span class='notice'>You cancel the spelll.</span>"
```
Comment (Henri215) [inline code/datums/spells/wizard_spells.dm:491]:
```suggestion
	selection_deactivated_message = "<span class='notice'>You cancel the spell.</span>"
```

File: code/datums/spells/wizard_spells.dm [hunk 12]
```diff
@@ -478,3 +478,48 @@
...
+	target.gib()
+	explosion(corpse_turf, 0, 0, 0, 0, silent = TRUE, breach = FALSE)
+	for(var/mob/living/M in range(4, corpse_turf))
+		if(M == user)
+			continue
+		var/range = get_dist_euclidian(M, corpse_turf)
+		range = max(1, range)
+		M.apply_damage(100 / range, BRUTE)
+		if(issilicon(M))
+			to_chat(M, "<span class='userdanger'>Your sensors are disabled and carapace ripped apart by the violent dark magic!</span>")
```
Comment (Henri215) [inline code/datums/spells/wizard_spells.dm:516]:
```suggestion
			to_chat(M, "<span class='userdanger'>Your sensors are disabled, and your carapace is ripped apart by the violent dark magic!</span>")
```

**Review by Henri215 — COMMENTED** [review]
image

Modified something here by accident?

File: code/datums/spells/wizard_spells.dm [hunk 13]
```diff
@@ -478,3 +478,48 @@
...
+/obj/effect/proc_holder/spell/corpse_explosion/create_new_targeting()
+	var/datum/spell_targeting/click/T = new
+	T.click_radius = 0
+	T.try_auto_target = FALSE
+	T.allowed_type = /mob/living
+	return T
+
+/obj/effect/proc_holder/spell/corpse_explosion/cast(list/targets, mob/user)
+	var/mob/living/target = targets[1]
+	if(!target || target.stat == !DEAD)
```
Comment (DGamerL) [inline code/datums/spells/wizard_spells.dm:503]:
```suggestion
	if(!target || target.stat != DEAD)
```

**Review by DGamerL — COMMENTED** [review]
Last thing

File: code/game/objects/effects/temporary_visuals/misc_visuals.dm [hunk 14]
```diff
@@ -103,6 +103,16 @@
 	icon_state = "liquify"
 	duration = 12
 
+/obj/effect/temp_visual/corpse_explosion
+	icon_state = "corpse_explosion"
+	icon = 'icons/effects/64x64.dmi'
+	icon_state = "corpse_explosion"
```
Comment (Henri215) [inline code/game/objects/effects/temporary_visuals/misc_visuals.dm:109]:
```suggestion
	icon = 'icons/effects/64x64.dmi'
	icon_state = "corpse_explosion"
```

**Review by Henri215 — COMMENTED** [review]
Everything else is lewcing good to me

**Comment by Henri215** [comment]
Datumizes spells got merged, time to merge master

**Comment by DGamerL** [comment]
!merge_upstream

**Review by S34NW — APPROVED** [review]

**Pull request merged**